notebooks/Binary Classification with the UCI Credit Card Payment Default Dataset.ipynb

```diff
@@ -0,0 +1,1150 @@
+    "\n",
+    "For this scenario, we use the [UCI dataset](https://archive.ics.uci.edu/ml/datasets/default+of+credit+card+clients) on credit card default payments in 2005 in Taiwan. Since the UCI dataset does not intrisically contain demographic disparities, we introduce a credit limit feature that is biased towards sex: this feature has predictive power for whether a female client will default, but not for male clients. This is one way in which demographic disparities can leak into the dataset despite removing sensitive demographic information.\n",
+    "\n",
+    "We begin by training a regular machine learning model on the biased data and show how the model performance differs for male and female clients. We use `fairlearn` to mitigate these differences using the `ThresholdOptimizer` and `GridSearch` techniques implemented in the package. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# General imports\n",
+    "import numpy as np\n",
+    "import pandas as pd\n",
+    "import matplotlib.pyplot as plt\n",
+    "%matplotlib inline\n",
+    "\n",
+    "# Data processing\n",
+    "from sklearn.model_selection import train_test_split\n",
+    "\n",
+    "# Models\n",
+    "import lightgbm as lgb\n",
+    "from sklearn.calibration import CalibratedClassifierCV\n",
+    "\n",
+    "# Fairlearn models and utils\n",
+    "from fairlearn.postprocessing import ThresholdOptimizer\n",
+    "from fairlearn.reductions import GridSearch, EqualizedOdds\n",
+    "from fairlearn.widget import FairlearnDashboard\n",
+    "\n",
+    "# Metrics\n",
+    "from fairlearn.metrics import group_selection_rate, group_zero_one_loss, group_recall_score, \\\n",
+    "                              metric_by_group, group_roc_auc_score\n",
+    "from sklearn.metrics import balanced_accuracy_score, accuracy_score, recall_score, roc_auc_score"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# The UCI Credit Card Payment Default Dataset\n",
+    "\n",
+    "The UCI datasets contains on 30,000 clients and their credit card transactions at a bank in Taiwan. In addition to static client features, the data contains the history of credit card bill payments between April and September 2005, as well as the balance limit of the client's credit card. The target is whether the client will default on a card payment in the following month, October 2005. One can imagine that a model trained on this data can be used in practice to determine whether a client is eligible for another product offering such as an auto loan. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [
```

Comment:
We usually remove outputs from all notebooks.

Reply:
Fixed!